### PR TITLE
feat: generated key pairs on installation

### PR DESCRIPTION
**Pull request by Arielpetit**
generated key pairs on installation

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,41 @@
...
+      await generateKeyPair();
+      console.log("Key pair generated successfully.");
+    };
+
+    // Listen for the appinstalled event
+    window.addEventListener("appinstalled", handleAppInstalled);
+
+    return () => window.removeEventListener("appinstalled", handleAppInstalled);
+  }, []);
+
```
Comment (NkwaTambe) [inline src/App.tsx:26]: These should not be in App.tsx

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,41 @@
...
   return (
     <Router>
-      <Routes>
-        {/* Define the routes */}
-        <Route path="/register" element={<Register />} />
-        <Route path="/otp" element={<OtpPage />} />
-        <Route path="/dashboard" element={<DashboardPage />} />
-      </Routes>
+      <div className="app">
```
Comment (NkwaTambe) [inline src/App.tsx:29]: Why  here ???

**Review by NkwaTambe — CHANGES_REQUESTED** [review]

**Review by NkwaTambe — COMMENTED** [review]
Can you implement a test for your components...

**Review by francis-pouatcha — APPROVED** [review]

File: src/services/keyManagement/generateKey.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as jose from "jose";
+
+async function generateKeyPair() {
+  // Generate an RSA key pair
```
Comment (francis-pouatcha) [inline src/services/keyManagement/generateKey.ts:4]: This is not an RSA key pair but an ec keypair

File: src/services/keyManagement/generateKey.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  const { publicKey, privateKey } = await jose.generateKeyPair("ES256", {
+    extractable: true,
+  });
+
+  // Convert keys to JSON format
+  const publicJWK = await jose.exportJWK(publicKey);
+  const privateJWK = await jose.exportJWK(privateKey);
+
+  console.log("Public Key:", publicJWK);
```
Comment (francis-pouatcha) [inline src/services/keyManagement/generateKey.ts:13]: Please do not log keys. Write tests to validate generated keys.

File: src/services/keyManagement/generateKey.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  console.log("Private Key:", privateJWK);
```
Comment (francis-pouatcha) [inline src/services/keyManagement/generateKey.ts:14]: Never log a private keys

**Review by francis-pouatcha — CHANGES_REQUESTED** [review]

**Comment by NkwaTambe** [comment]
Please fix the failing test

File: jest.config.js [hunk 6]
```diff
@@ -1 +1,25 @@
-setupFilesAfterEnv: ['<rootDir>/jest-setup.js']
\ No newline at end of file
+/* eslint-disable */
+export default {
+    displayName: 'contact-service',
+    testEnvironment: 'node',
```
Comment (NkwaTambe) [inline jest.config.js]: The test environtment has been specified already in the `vitest.config.ts`....
we are using `jsdom` as our test environtment

File: package.json [hunk 7]
```diff
@@ -15,20 +15,24 @@
     "coverage": "vitest run --coverage"
   },
   "dependencies": {
-    "@types/jest": "^29.5.14",
```
Comment (NkwaTambe) [inline package.json:18]: we need these `types`..

`@types/jest` provides us we functions like `expect` which we use in our test

File: src/pages/RegisterPage.tsx [hunk 8]
```diff
@@ -58,14 +60,15 @@ const Register: React.FC = () => {
       alert("Please enter a valid phone number.");
       return;
     }
```
Comment (NkwaTambe) [inline src/pages/RegisterPage.tsx:62]: Please where are you using these type in your code ??
i mean the type `CountryOption`

File: src/pages/RegisterPage.tsx [hunk 9]
```diff
@@ -58,14 +60,15 @@ const Register: React.FC = () => {
...
+
     setIsLoading(true); // Set loading state
     try {
-      // Replace with your actual OTP sending logic
       console.log("Sending OTP to:", phoneNumber);
-      // Simulate a network request with setTimeout
+      console.log("Generating key pair...");
```
Comment (NkwaTambe) [inline src/pages/RegisterPage.tsx:67]: please remove the `console.log`

**Review by NkwaTambe — CHANGES_REQUESTED** [review]

File: src/pages/RegisterPage.tsx (hunk 8 above)
Comment (Arielpetit) [reply src/pages/RegisterPage.tsx:61]: line 18
line 19
line 26
line 30

**Review by Elwizzy12 — DISMISSED** [review]

**Review by Koufan-De-King — APPROVED** [review]

**Comment by Koufan-De-King** [comment]
This ticket was resolved in 
- [ ] #42

**Pull request closed without merging**